src/geometry/vector.h

```diff
@@ -291,6 +299,13 @@ Vector3<T> normalize(const Vector3<T> &v) {
   return v / v.length();
 }
 
+template <typename T>
+inline std::ostream &operator<<(std::ostream &stream, const Vector3<T> &v) {
```

Comment:
You didn't include <ostream> for this to work in case it gets removed in quaternion.h. Would be good to have it in vector.h

Reply:
Fixed in 1df931c